cmake/coveralls.cmake

```diff
@@ -62,7 +62,6 @@ function(code_coverage _COVERAGE_SRCS _COVERALLS_UPLOAD _CMAKE_SCRIPT_PATH)
 endfunction()
 
 if(WITH_COVERAGE)
```

Comment:
Why unset `CMAKE_BUILD_TYPE` from `Debug`? You should analysis the fail unit-test in #18540 one by one.